tests/unittests/sources/test_azure.py

```diff
@@ -4487,6 +4654,37 @@ def test_imds_failure_results_in_provisioning_failure(self):
         assert len(self.mock_kvp_report_success_to_host.mock_calls) == 0
 
 
+class TestCheckAzureProxyAgent:
+    @pytest.fixture(autouse=True)
+    def proxy_setup(self, azure_ds, mock_subp_subp):
+        self.azure_ds = azure_ds
```

Comment:
since all the tests use caplog too, perhaps include it here too?

Reply:
done